pypiserver/templates/deployment.yaml

```
@@ -47,7 +47,7 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           command: ["pypi-server"]
           args:
-            - --passwords=/config/.htpasswd
+            - --passwords={{ if .Values.auth.actions }}/config/.htpasswd{{ else }}.{{ end }}
             - --authenticate={{ .Values.auth.actions }}
```

It looks like you need to pass `.` also when you want unauthenticated calls (https://pypi.org/project/pypiserver/#quickstart-installation-and-usage)
```suggestion
            - --authenticate={{ .Values.auth.actions | default "." }}
```